docs/upgrade.md

```diff
@@ -3,5 +3,263 @@ title: Upgrade guide
 description: Guide to update between major Powertools for AWS Lambda (TypeScript) versions
 ---
 
-!!! warning
-    This guide is a work in progress. We'll update it as we get closer to the 2.0 release. If you have any questions, or want to follow the progress, please join the discussion on the [GitHub issue](https://github.com/aws-powertools/powertools-lambda-typescript/issues/1714).
\ No newline at end of file
+## Migrate from v1 to v2
+
+To make your transition to v2 as smooth as possible while still providing you with the most up-to-date features and patterns we have made targeted breaking changes to some of the utilities. This guide will help you migrate your existing codebase to v2.
+
+### Quick summary
+
+Below a list of the main changes introduced in v2, review the sections below for more details and code examples for the changes that apply to your codebase.
+
+| Area                                   | Change                                                                                                                                           | Code change required |
+| -------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------ | -------------------- |
+| **ES Modules support**                 | Added ES Modules support via dual CommonJS and ES Module bundling, this enables top-level `await` and tree-shaking.                              | -                    |
+| **Middy.js middleware imports**        | Updated import path for Middy.js middlewares to leverage subpath exports - i.e. `@aws-lambda-powertools/tracer/middleware`.                      | Yes                  |
+| **Types imports**                      | Updated import path for TypeScript types to leverage subpath exports - i.e. `@aws-lambda-powertools/logger/types`.                               | Yes                  |
+| **Logger - log sampling**              | Changed implementation of [log sampling](./core/logger.md#sampling-logs) to dynamically switch log level to `DEBUG` on a percentage of requests. | -                    |
```

Comment:
```suggestion
| **Logger**              | Changed [log sampling](./core/logger.md#sampling-logs) to dynamically switch log level to `DEBUG` on a percentage of requests. | -                    |
```

You are repeating the `change` in the `area` field, making it longer to read at a quick glance. By keeping `Logger`, it's easier for the eye to quickly scan how many breaking changes `Logger` have (estimate/perception) - if they use it, they can then read the actual `change`.

At first, it can look like a win by having the area with the change so they can skip ahead, but what we're actually optimizing in the `Quick summary` section is for people to quickly scan what major areas are having breaking changes (`log sampling` being a minor area).

line editing to cut unnecessary words.